leader/leader.go

```diff
@@ -283,46 +300,47 @@ func (l *Client) StepDown() {
 
 // getFirstValue returns the current value for key if it exists, or waits
 // for the value to appear and loops until client.Close is called
-func (l *Client) getFirstValue(key string, retryPeriod time.Duration) (*client.Response, error) {
+func (l *Client) getFirstValue(key string, retryPeriod time.Duration) *client.Response {
```

Comment:
I'm also a bit puzzled about `retry` parameter - how does the caller know the best value to select? Why not just try until either the context expires or the client is explicitly closed? What happens if the retry duration is too fast or too slow?

Reply:
This value gets propagated from the caller and depends on the election term the caller selects.  I can add a context here but I'd like to avoid changing the logic heavily at this time.